src/datapump/jobs/version_update.py

```diff
@@ -0,0 +1,186 @@
+from enum import Enum
+from typing import List, Optional, Union, Dict
+
+from pydantic import StrictInt
+
+from ..clients.data_api import DataApiClient
+from ..jobs.jobs import Job, JobStatus
+from ..util.exceptions import DataApiResponseError
+
+
+class NonNumericFloat(str, Enum):
+    nan = "nan"
+
+
+NoDataType = Union[StrictInt, NonNumericFloat]
+
+
+class VersionUpdateJobStep(str, Enum):
+    starting = "starting"
+    creating_tile_set = "creating_tile_set"
+    creating_tile_cache = "creating_tile_cache"
+    creating_aux_assets = "creating_aux_assets"
+    mark_latest = "mark_latest"
+
+
+class VersionUpdateJob(Job):
+    dataset: str
+    version: str
+    source_uri: List[str]
+    calc: Optional[str]
+    grid: str
+    max_zoom: int
+    data_type: str
+    no_data: Optional[Union[List[NoDataType], NoDataType]]
+    pixel_meaning: str
+    tile_cache_symbology: Optional[Dict[str, str]]
+
+    def next_step(self):
+        if self.step == VersionUpdateJobStep.starting:
```

Comment:
You don't have to change this now, but I think we might need to refactor in general how we do next_step() in the future. The giant if-else branches don't feel super right.  Probably we should make each step an independent class, where each class check lets you know when it's done. And then each job is just a list of step classes to execute in order.

Reply:
Yes, I was thinking something similar but didn't want to rock the boat too much. Totally on-board with that approach.

Reply:
Agree too. for large if/ else blocks on the same variable, I generally prefer a dictionary lookup. 
This make the code more compact and also faster to execute.